erpnext/hr/doctype/employee_grievance/employee_grievance.js

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2021, Frappe Technologies Pvt. Ltd. and contributors
+// For license information, please see license.txt
+
+frappe.ui.form.on('Employee Grievance', {
+	setup: function(frm) {
+		frm.set_query('grievance_against_party', function() {
+			return {
+				filters: {
+					name: ['in', [
+						'Company', 'Department', 'Employee Group', 'Employee Grade', 'Employee']
+					]
+				}
+			};
+		});
+		frm.set_query('associated_document_type', function() {
+			let ignore_modules = ["Setup", "Core", "Integrations", "Automation", "Website",
+				"Utilities", "Event Streaming", "Social", "Chat", "Data Migration", "Printing", "Desk", "Custom"];
+			return {
+				filters: {
+					istable: 0,
+					issingle: 0,
+					module: ["Not In", ignore_modules]
+				}
+			};
+		});
+	},
+
+	refresh: function(frm) {
```

Comment:
unnecessary client side code

Reply:
this should be implemented via workflow

Reply:
these are good to have features when dealing with grievances. It's just code to add custom buttons for additional salary creation or changing employee status to Suspended. Employees can be got suspended for Employee Grievance like Employee Abuse, bullying. 

I am not sure how workflow doctype can handle document action mapping?
It just allows setting status or updating some value in the same document

Reply:
Based on the selection of employee made in the "Raised By" field, the selected employee name should be refrained from getting reflected in the dropdown of "Grievance Against" field. 

![Employee Grievance ](https://user-images.githubusercontent.com/32797974/122364940-debe3f80-cf77-11eb-9c3b-9176f91d3e94.png)

Reply:
fixed